libs/libcommon/src/libcommon/parquet_utils.py

```diff
@@ -252,6 +248,8 @@ def query(self, offset: int, length: int) -> pa.Table:
         with StepProfiler(
             method="parquet_index_with_metadata.query", step="load the remote parquet files using metadata from disk"
         ):
+            httpfs = HTTPFileSystem()
+            session = asyncio.run(httpfs.set_session())
```

Comment:
This can be a global singleton that is instantiated the first time `query` is called.

This way we reuse the same session across queries and reduce the latency :)

Reply:
Sure, I moved it to HTTPFileSystemSession